vllm/model_executor/models/minicpmv.py

```diff
@@ -559,119 +510,72 @@ def get_image_bounds(self, input_ids: torch.Tensor):
-                all_pixel_values = [
-                    i.flatten(end_dim=1).permute(1, 0) for i in pixel_values
-                ]
-                if all_pixel_values:
-                    tgt_sizes = torch.vstack(tgt_sizes).type(torch.int32)
-                    max_patches = torch.max(tgt_sizes[:, 0] * tgt_sizes[:, 1])
-                    all_pixel_values = torch.nn.utils.rnn.pad_sequence(
-                        all_pixel_values, batch_first=True, padding_value=0.0)
-                    B, L, _ = all_pixel_values.shape
-                    all_pixel_values = all_pixel_values.permute(
-                        0, 2, 1).reshape(B, 3, -1, L)
-                    patch_attn_mask = torch.zeros((B, 1, max_patches),
-                                                  dtype=torch.bool,
-                                                  device=device)
-                    if self.version == (2, 5):
-                        for i in range(B):
-                            patch_attn_mask[i, :tgt_sizes[i][0] *
-                                            tgt_sizes[i][1]] = True
-                        vision_embedding = self.vpm(
-                            all_pixel_values.type(dtype),
-                            patch_attention_mask=patch_attn_mask
-                        ).last_hidden_state
-                    else:
-                        for i in range(B):
-                            patch_attn_mask[i, 0, :tgt_sizes[i][0] *
-                                            tgt_sizes[i][1]] = True
-                        vision_embedding = self.vpm(
-                            all_pixel_values.type(dtype),
-                            patch_attention_mask=patch_attn_mask,
-                            tgt_sizes=tgt_sizes).last_hidden_state
-
-                    vision_hidden_states = self.resampler(
-                        vision_embedding, tgt_sizes)
-
-                else:  # no image
-                    dummy_feature = []
-                    vision_hidden_states = dummy_feature
+    def get_embedding(
+        self,
+        input_ids: torch.Tensor,
+        image_inputs: Optional[MiniCPMVImageInputs],
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        if image_inputs is None:  # No image
+            vlm_embedding = torch.tensor([], device=input_ids.device)
+            vision_hidden_states = torch.tensor([], device=input_ids.device)
         else:
-            vision_hidden_states = data["vision_hidden_states"]
-
-        return vision_hidden_states
-
-    def get_embedding(self, data: Dict[str, Union[List[torch.Tensor],
-                                                  torch.Tensor]]):
-        input_ids = data["input_ids"]
-
-        vision_hidden_states = self.get_vision_hidden_states(data)
-        if vision_hidden_states is not None and len(vision_hidden_states) > 0:
+            vision_hidden_states = self.get_vision_hidden_states(image_inputs)
             image_bounds = self.get_image_bounds(input_ids)
```

Comment:
From here starts, these lines should be outside of `else`.

Reply:
@jeejeelee Check here. `image_bounds` calculating and `input_ids` embedding should be always done.

Reply:
See https://github.com/vllm-project/vllm/pull/7020#issuecomment-2267079891